helpers/auth/authentication.py

```diff
@@ -82,8 +81,7 @@ def save_user(self):
                 try:
                     user_data = User(email=email, name=name, picture=picture)
                     user_data.save()
-                    user_role = UsersRole(
-                        user_id=user_data.id, role_id=role.id)
+                    user_role = user_data.roles.append(role)
                     user_role.save()
```

Comment:
@dnuwa I think variable naming in these lines of code can cause a bit of confusion, please refactor. There is also a bit of an issue with the logical flow here; why are you saving the user_data object twice for example?